homeassistant/components/light/decora.py

```diff
@@ -52,15 +53,21 @@ def __init__(self, device):
         """Initialize the light."""
         # pylint: disable=import-error
         import decora
-
         self._name = device['name']
         self._address = device['address']
         self._key = device["key"]
         self._switch = decora.decora(self._address, self._key)
-        self._switch.connect()
+        initial = time.time()
+        while True:
+            if time.time() - initial >= 10:
+                return None
+            try:
+                self._switch.connect()
+                break
+            except decora.decoraException:
+                pass
```

Comment:
Can you move that out of `__init__` and check it on setup. Maby if no connection is possible, it will not add the entity. If you do this on setup, it block/slow done also the hole startup. Maby you add this with a EVENT_HOMEASSISTANT_START.